test/tools/ossfuzz/yulProto.proto

```diff
@@ -276,7 +289,13 @@ message Block {
 }
 
 message Function {
-  required Block statements = 1;
+  required Block statements  = 1;
+  required int32 inparams   = 2;
+  required int32 outparams  = 3;
+}
+
+message Program {
```

Comment:
Shouldn't `Program` be identical to `Block`?

Reply:
Ah, `Statement` does not include `Function` - is that on purpose?

Reply:
`MultiVarAssignment` and `MultiVarDecl`  necessarily include `Function` calls. So, indirectly `Function` call is a statement. `Function` denotes function definition.

Reply:
I'm saying that function definition should be a statement - it can occur almost anywhere in the code.

Reply:
ah, sorry. you're right. Function definition can be a statement. The reason I used the current message hierarchy is to register in advance the list of functions that may later be called.

Changing it may require some code refactoring that I haven't fully thought about. Essentially, we may not know in advance which functions to call until they are visited. I'm also not sure what scoping rules apply to functions. Can they be called before they are defined e.g., 

```
mstore(0, foo())
if expression define foo() -> a { a := 42}
```

Reply:
Scoping rules are the same as we have for variables, just that you can call functions before they are defined. You cannot call them if they are defined in nested scopes, only if they are defined in the current or in outer scopes.

I'm fine with restricting the fuzzer to a certain layout, since the function hoister will pull out all functions rather soon anyway.